ietf/utils/migrations/0002_convert_timestamps_to_utc.py

```diff
@@ -178,92 +213,93 @@ class Migration(migrations.Migration):
-UPDATE nomcom_topicfeedbacklastseen SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE oidc_provider_code SET expires_at = CONVERT_TZ(expires_at, 'PST8PDT', 'UTC');
-UPDATE oidc_provider_token SET expires_at = CONVERT_TZ(expires_at, 'PST8PDT', 'UTC');
-UPDATE oidc_provider_userconsent SET date_given = CONVERT_TZ(date_given, 'PST8PDT', 'UTC');
-UPDATE oidc_provider_userconsent SET expires_at = CONVERT_TZ(expires_at, 'PST8PDT', 'UTC');
-UPDATE person_email SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE person_historicalemail SET history_date = CONVERT_TZ(history_date, 'PST8PDT', 'UTC');
-UPDATE person_historicalemail SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE person_historicalperson SET history_date = CONVERT_TZ(history_date, 'PST8PDT', 'UTC');
-UPDATE person_historicalperson SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE person_person SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE person_personalapikey SET created = CONVERT_TZ(created, 'PST8PDT', 'UTC');
-UPDATE person_personalapikey SET latest = CONVERT_TZ(latest, 'PST8PDT', 'UTC');
-UPDATE person_personevent SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE request_profiler_profilingrecord SET end_ts = CONVERT_TZ(end_ts, 'PST8PDT', 'UTC');
-UPDATE request_profiler_profilingrecord SET start_ts = CONVERT_TZ(start_ts, 'PST8PDT', 'UTC');
-UPDATE review_historicalreviewassignment SET assigned_on = CONVERT_TZ(assigned_on, 'PST8PDT', 'UTC');
-UPDATE review_historicalreviewassignment SET completed_on = CONVERT_TZ(completed_on, 'PST8PDT', 'UTC');
-UPDATE review_historicalreviewassignment SET history_date = CONVERT_TZ(history_date, 'PST8PDT', 'UTC');
-UPDATE review_historicalreviewersettings SET history_date = CONVERT_TZ(history_date, 'PST8PDT', 'UTC');
-UPDATE review_historicalreviewrequest SET history_date = CONVERT_TZ(history_date, 'PST8PDT', 'UTC');
-UPDATE review_historicalreviewrequest SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE review_historicalunavailableperiod SET history_date = CONVERT_TZ(history_date, 'PST8PDT', 'UTC');
-UPDATE review_reviewassignment SET assigned_on = CONVERT_TZ(assigned_on, 'PST8PDT', 'UTC');
-UPDATE review_reviewassignment SET completed_on = CONVERT_TZ(completed_on, 'PST8PDT', 'UTC');
-UPDATE review_reviewrequest SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE review_reviewwish SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE south_migrationhistory SET applied = CONVERT_TZ(applied, 'PST8PDT', 'UTC');
-UPDATE submit_preapproval SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE submit_submissioncheck SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE submit_submissionevent SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
-UPDATE tastypie_apikey SET created = CONVERT_TZ(created, 'PST8PDT', 'UTC');
-UPDATE utils_versioninfo SET time = CONVERT_TZ(time, 'PST8PDT', 'UTC');
+UPDATE auth_user SET date_joined = CONVERT_TZ(date_joined, 'PST8PDT', 'UTC') WHERE date_joined >= '1969-12-31 16:00:01';
```

Comment:
Does the where clause have a runtime impact? We're only adding it for documentation (and maybe a tiny amount of protection against the contract with CONVERT_TZ changing, but that's vanishingly small isn't it?) since the effect of CONVERT_TZ only operates on times already in that clause anyhow - if this makes the migration time significantly longer, I would address it with a comment instead?

Reply:
I didn't notice a difference - it looks like it adds 2 seconds to my dev machine where the baseline migration takes 21 seconds. I'm inclined to keep it but don't feel strongly.

#### Without `WHERE`: 21s
```
dev ➜ /workspace $ date; ietf/manage.py migrate utils 0002_convert_timestamps_to_utc; date
Wed 07 Sep 2022 05:27:34 PM UTC
Operations to perform:
  Target specific migration: 0002_convert_timestamps_to_utc, from utils
Running migrations:
  Applying utils.0002_convert_timestamps_to_utc... OK
Wed 07 Sep 2022 05:27:55 PM UTC
```

#### With `WHERE`: 23s
```
dev ➜ /workspace $ date; ietf/manage.py migrate utils 0002_convert_timestamps_to_utc; date
Wed 07 Sep 2022 05:39:23 PM UTC
Operations to perform:
  Target specific migration: 0002_convert_timestamps_to_utc, from utils
Running migrations:
  Applying utils.0002_convert_timestamps_to_utc... OK
Wed 07 Sep 2022 05:39:46 PM UTC
```

Reply:
its not the 20 minutes I feared it might be - so keeping it is fine.